src/main/groovy/org/scoverage/CoverageChecker.groovy

```diff
@@ -41,43 +38,22 @@ enum CoverageType {
 /**
  * Throws a GradleException if overall coverage dips below the configured percentage.
  */
-@CacheableTask
-class OverallCheckTask extends DefaultTask {
-
-    /** Type of coverage to check. Available options: Line, Statement and Branch */
-    @Input
-    final Property<CoverageType> coverageType = project.objects.property(CoverageType)
-    @Input
-    final Property<BigDecimal> minimumRate = project.objects.property(BigDecimal)
-
-    @Input
-    final Property<File> reportDir = project.objects.property(File)
+class CoverageChecker {
 
-    /** Overwrite to test for a specific locale. */
-    @Input
-    final Property<Locale> locale = project.objects.property(Locale).value(Locale.getDefault())
+    final Logger logger
 
-    @TaskAction
-    void requireLineCoverage() {
-        NumberFormat nf = NumberFormat.getInstance(locale.get())
-
-        Exception failure = checkLineCoverage(nf, reportDir.get(), coverageType.get(), minimumRate.get().doubleValue())
-
-        if (failure) throw failure
+    CoverageChecker(Logger logger) {
+        this.logger = logger
     }
 
-    @VisibleForTesting
-    protected static String errorMsg(String actual, String expected, CoverageType type) {
-        "Only $actual% of project is covered by tests instead of $expected% (coverageType: $type)"
+    public Exception checkLineCoverage(File reportDir, CoverageType coverageType, double minimumRate) throws GradleException {
```

Comment:
this seems a little odd - is there any intention to actually return an exception here?

Reply:
Just leftover from previous code :) I'm surprised this method compiles (`return void`).